[CI] `poetry-build-test` fails for external branches

PRs created by external contributors fail on `poetry-build-test` since it tries checking out branch from main skypilot repo, not the PR's source repo. [Example run](https://github.com/skypilot-org/skypilot/actions/runs/6131701431/job/16641911660?pr=2242).

```
Failed to clone https://github.com/skypilot-org/skypilot.git at 'derek-falcon-example', verify ref exists on remote.
Error: Process completed with exit code 1.
```

This line probably needs to be updated to use the PR's source repo:
https://github.com/skypilot-org/skypilot/blob/e164dc29fc9aee2b14c4abe634af9b3b84e9a806/.github/workflows/test-poetry-build.yml#L39

--- a/.github/workflows/test-poetry-build.yml
+++ b/.github/workflows/test-poetry-build.yml
@@ -36,7 +36,7 @@ jobs:
           python = "3.10.x"
 
           [tool.poetry.group.dev.dependencies]
-          skypilot = {git = "https://github.com/skypilot-org/skypilot.git", branch = "${{ github.head_ref }}"}
+          skypilot = {git = "${{ github.event.pull_request.head.repo.html_url }}", branch = "${{ github.head_ref }}"}
 
           [build-system]
           requires = ["poetry-core"]

--- a/.github/workflows/test-poetry-build.yml
+++ b/.github/workflows/test-poetry-build.yml
@@ -36,7 +36,7 @@ jobs:
           python = "3.10.x"
 
           [tool.poetry.group.dev.dependencies]
-          skypilot = {git = "${{ github.event.pull_request.head.repo.html_url }}", branch = "${{ github.head_ref }}"}
+          skypilot = {git = "${{ github.event.pull_request.head.repo.html_url }}.git", branch = "${{ github.head_ref }}"}
 
           [build-system]
           requires = ["poetry-core"]

